tagstudio/src/qt/modals/ffmpeg_checker.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import math
+from pathlib import Path
+import subprocess
+
+from PIL import Image, ImageQt
+from pydub.utils import which
+from PySide6.QtCore import Signal, Qt, QUrl
+from PySide6.QtGui import QPixmap, QDesktopServices
+from PySide6.QtWidgets import QMessageBox
+
+
+class FfmpegChecker(QMessageBox):
+    """A warning dialog for if FFmpeg is missing."""
+
+    help_url = "https://github.com/TagStudioDev/TagStudio/blob/Alpha-v9.4/doc/utilities/ffmpeg.md"
+
+    def __init__(self):
+        super().__init__()
+
+        self.setIcon(QMessageBox.Warning)
+
+        self.setText("Warning: Could not find FFmpeg installation")
+        self.setStandardButtons(
+            QMessageBox.Help | QMessageBox.Ignore | QMessageBox.Cancel
+        )
+        self.setDefaultButton(QMessageBox.Ignore)
+        # Enables the cancel button but hides it to allow for click X to close dialog
+        self.button(QMessageBox.Cancel).hide()
+
+        # Blocks other application interactions until resolved
+        self.setWindowModality(Qt.ApplicationModal)
+        self.setWindowTitle("Warning: Missing dependency")
+
+        self.ffmpeg_missing = True
+        self.ffprobe_missing = True
+
+    def installed(self):
+        """Checks if both FFmpeg and FFprobe are installed and in the PATH."""
+        # Same checker that ffmpeg-python uses
+        if which("ffmpeg"):
+            self.ffmpeg_missing = False
+            logging.info(f"FFmpeg found!")
+        if which("ffprobe"):
+            self.ffprobe_missing = False
+            logging.info(f"FFprobe found!")
+        # Reverse from missing to installed
+        return not (self.ffmpeg_missing or self.ffprobe_missing)
```

Comment:
Is there a logical or stylistic reason for checking negatives? wouldn't it be better to have it called `self.ffmpeg` and have it to `False` until you find it in `PATH` and set it to True, chance not needing to do any inversion in the return statement?

I know that I am sharing my own style here, and different styles are fine, I have just not seen this to many times so hence my question.

Reply:
No, I'm not sure why I flip-floppde the logic there, I had it the other way around previously. 🙃 Fixed -sorry